Web.Client/Pages/Admin/GradeRegistrationCriteriaList.razor.cs

```diff
@@ -0,0 +1,47 @@
+﻿using MensaGymnazium.IntranetGen3.Contracts;
+using MensaGymnazium.IntranetGen3.Web.Client.Services.DataStores;
+
+namespace MensaGymnazium.IntranetGen3.Web.Client.Pages.Admin;
+
+public partial class GradeRegistrationCriteriaList
+{
+	[Inject] protected IGradeFacade GradeFacade { get; set; }
+	[Inject] protected IHxMessengerService Messenger { get; set; }
+	[Inject] protected IGradesDataStore GradesDataStore { get; set; }
+
+	private HxGrid<GradeRegistrationCriteriaDto> gridComponent; // @ref
+	private GradeRegistrationCriteriaEdit editComponent; // @ref
+	private GradeRegistrationCriteriaDto itemInEdit = new();
+	private GradeRegistrationCriteriaDto itemSelected;
+
+	protected override async Task OnInitializedAsync()
+	{
+		await GradesDataStore.EnsureDataAsync();
+	}
+
+	private async Task<GridDataProviderResult<GradeRegistrationCriteriaDto>> GetData(GridDataProviderRequest<GradeRegistrationCriteriaDto> request)
+	{
+		var data = await GradeFacade.GetGradeRegistrationCriteriasAsync(request.CancellationToken);
+		return request.ApplyTo(data);
+	}
+
+	private string GetGradeName(int gradeId)
+	{
+		// Xopa: Throws error when I don't make it nullable, but that makes no sense.
+		// There should always be all the grades we're asking for...
+		return GradesDataStore.GetByKeyOrDefault(gradeId)?.Name;
```

Comment:
`await GradesDataStore.EnsureDataAsync();` is async
which means the very first rendering happens BEFORE the task completes
Sakra, proč ti to píšu anglicky.
Trochu vysvětlený je to tady: https://learn.microsoft.com/en-us/aspnet/core/blazor/components/lifecycle?view=aspnetcore-8.0
Ultimátně tady: https://github.com/dotnet/aspnetcore/blob/d28d0c597c1a00fe25353ee4653fab49da404c2f/src/Components/Components/src/ComponentBase.cs#L232-L245